src/main/java/com/parkhomovsky/bookstore/service/implementation/OrderServiceImpl.java

```diff
@@ -0,0 +1,153 @@
+package com.parkhomovsky.bookstore.service.implementation;
+
+import com.parkhomovsky.bookstore.dto.order.OrderDto;
+import com.parkhomovsky.bookstore.dto.order.OrderPlaceRequestDto;
+import com.parkhomovsky.bookstore.dto.order.OrderUpdateStatusRequest;
+import com.parkhomovsky.bookstore.dto.order.StatusUpdateResponseDto;
+import com.parkhomovsky.bookstore.dto.orderitem.OrderItemDto;
+import com.parkhomovsky.bookstore.enums.Status;
+import com.parkhomovsky.bookstore.exception.EmptyShoppingCartException;
+import com.parkhomovsky.bookstore.exception.EntityNotFoundException;
+import com.parkhomovsky.bookstore.mapper.OrderItemsMapper;
+import com.parkhomovsky.bookstore.mapper.OrderMapper;
+import com.parkhomovsky.bookstore.model.CartItem;
+import com.parkhomovsky.bookstore.model.Order;
+import com.parkhomovsky.bookstore.model.OrderItem;
+import com.parkhomovsky.bookstore.model.ShoppingCart;
+import com.parkhomovsky.bookstore.model.User;
+import com.parkhomovsky.bookstore.repository.order.OrderRepository;
+import com.parkhomovsky.bookstore.repository.orderitems.OrderItemsRepository;
+import com.parkhomovsky.bookstore.service.OrderService;
+import com.parkhomovsky.bookstore.service.ShoppingCartService;
+import java.math.BigDecimal;
+import java.time.LocalDateTime;
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Collectors;
+import lombok.RequiredArgsConstructor;
+import org.springframework.data.domain.Pageable;
+import org.springframework.security.core.Authentication;
+import org.springframework.stereotype.Service;
+
+@Service
+@RequiredArgsConstructor
+public class OrderServiceImpl implements OrderService {
+    private final ShoppingCartService shoppingCartService;
+    private final OrderRepository orderRepository;
+    private final OrderItemsRepository orderItemsRepository;
+    private final OrderMapper orderMapper;
+    private final OrderItemsMapper orderItemsMapper;
+
+    public OrderDto process(OrderPlaceRequestDto orderPlaceRequestDto,
+                            Authentication authentication) {
+        Set<OrderItem> orderItems = getOrderItemsFromShoppingCart();
+        Order order = buildOrder(orderPlaceRequestDto.getShippingAddress(),
+                orderItems, authentication);
+        orderRepository.save(order);
+        List<OrderItemDto> orderItemDtos = orderItems.stream()
+                .map(orderItemsMapper::toDto)
+                .collect(Collectors.toList());
+        orderItems.forEach(orderItem -> orderItem.setOrder(order));
+        OrderDto orderDto = buildOrderDto(order, orderItemDtos);
+        orderItemsRepository.saveAll(orderItems);
+        return orderDto;
+    }
+
+    @Override
+    public List<OrderDto> getAll(Pageable pageable, Authentication authentication) {
+        User currentUser = (User) authentication.getPrincipal();
```

Comment:
this logic should not be here, and getAll method should not accept authentication. Whether accept userId as a param or extract user fetching to user related services (fix it in all places)